fedimint-rocksdb/src/lib.rs

```diff
@@ -115,6 +115,21 @@ impl<'a> IDatabaseTransactionOps<'a> for RocksDbTransaction<'a> {
     }
 
     async fn raw_find_by_prefix(&mut self, key_prefix: &[u8]) -> Result<PrefixStream<'_>> {
+        // turn an `iter` into a `Stream` where every `next` is ran inside
+        // `block_in_place` to offload the blocking calls
+        fn convert_to_async_stream<'i, I>(iter: I) -> impl futures::Stream<Item = I::Item>
+        where
+            I: Iterator + Send + 'i,
+            I::Item: Send,
+        {
+            stream::unfold(iter, |mut iter| async move {
+                tokio::task::block_in_place(move || {
+                    let item = iter.next();
+                    item.map(|item| (item, iter))
```

Comment:
Why doesn't this need to be a `move` closure? How is `iter` getting in there by value or isn't it? :open_mouth:

Reply:
Yeah. I haven't noticed and I don't understand. Must be some auto-convenience thing, or something being `Copy` here?

Anyway, changed to `move` as it feels more ... proper.

Reply:
TIL: apparently using something by-value in a closure moves it automatically and the `move` keyword is only useful when variables are used by-ref but one still wants them to move into the closure for lifetime issues.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d81e7e6dc74c49ddaeef283a63d20d01